src/convertToObject.js

```diff
@@ -6,7 +6,35 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  // Transform string into array of lines
+  const propertiesArray = sourceString.split(';');
+  const styles = {};
+
+  // Transform separate lines of the array of properties into key-value pairs
+  propertiesArray.forEach((property) => {
+    const parts = property.split(':');
+
+    const key = dataCleanup(parts[0]);
+    const value = dataCleanup(parts[1]);
+
+    // Add to the resulting object only the lines with both key and value
+    if (key.length && value.length) {
+      styles[key] = value;
+    }
+  });
```

Comment:
[CODE STYLE] - All chained methods called on the same level should have the same indentation to improve readability. In this case, you can chain the `split`, `map`, and `filter` methods to transform the string into an array of valid style objects without using a loop.